numpy/lib/shape_base.py

```diff
@@ -405,18 +404,23 @@ def array_split(ary, indices_or_sections, axis=0):
         Ntotal = len(ary)
     try:
         # handle scalar case.
-        Nsections = len(indices_or_sections) + 1
-        div_points = [0] + list(indices_or_sections) + [Ntotal]
+        Nsections = len(chunks) + 1
+        div_points = [0] + list(chunks) + [Ntotal]
     except TypeError:
-        # indices_or_sections is a scalar, not an array.
-        Nsections = int(indices_or_sections)
-        if Nsections <= 0:
-            raise ValueError('number sections must be larger than 0.')
-        Neach_section, extras = divmod(Ntotal, Nsections)
-        section_sizes = ([0] +
-                         extras * [Neach_section+1] +
-                         (Nsections-extras) * [Neach_section])
-        div_points = _nx.array(section_sizes).cumsum()
+        # chunks is not iterable
+        try:
```

Comment:
Sorry, I thought it was clear: that bit of code takes care of when 'indices_or_sections' or 'chunks' is neither an int nor an iterable (notably None). In that case the behavior, which make more sense to me, is to split into the maximum number of chunks.